From: Fabian Frederick <fabf@skynet.be>
Date: Fri, 27 Jun 2014 22:51:52 +0200
Subject: [PATCH 165/223] drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c:
 remove unnecessary null test before debugfs_remove_recursive
Patch-mainline: v3.17-rc1
Git-commit: 9f16dc2ec7cb3527c66581ad762876ba1f774cdb
References: bsc#909577 FATE#317550

Fix checkpatch warning:
"WARNING: debugfs_remove_recursive(NULL) is safe this check is probably not required"

Cc: Hariprasad S <hariprasad@chelsio.com>
Cc: netdev@vger.kernel.org
Signed-off-by: Fabian Frederick <fabf@skynet.be>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
@@ -6532,8 +6532,7 @@ static void remove_one(struct pci_dev *p
 			if (adapter->port[i]->reg_state == NETREG_REGISTERED)
 				unregister_netdev(adapter->port[i]);
 
-		if (adapter->debugfs_root)
-			debugfs_remove_recursive(adapter->debugfs_root);
+		debugfs_remove_recursive(adapter->debugfs_root);
 
 		/* If we allocated filters, free up state associated with any
 		 * valid filters ...
